### PR TITLE
Also stop execution if PreventPvPEvent is cancelled

### PR DESCRIPTION
**Pull request by FrankHeijden**

### CODE REVIEW

File: src/main/java/me/ryanhamshire/GriefPrevention/EntityEventHandler.java [hunk 1]
```diff
@@ -1413,7 +1413,6 @@ else if (GriefPrevention.instance.config_pvp_noCombatInPlayerLandClaims || Grief
                         {
                             event.setIntensity(effected, 0);
                             GriefPrevention.sendMessage(thrower, TextMode.Err, Messages.PlayerInPvPSafeZone);
-                            continue;
```
Comment (FrankHeijden) [inline src/main/java/me/ryanhamshire/GriefPrevention/EntityEventHandler.java:1447]: This statement was useless anyway since it was at the end of the loop

**Comment by RoboMWM** [comment]
Not really familiar with the code here, so what's the existing behavior and what is it being modified to?

**Comment by FrankHeijden** [comment]
Currently the code returns when the preventPVPEvent is not called (whichhappens when there are no plugins modify this event), it makes sense when the event is cancelled to also stop there, to disallow GP from taking other actions.

**Comment by RoboMWM** [comment]
Ok so I'm not sure if that's really what you're doing, as it looks more like you're just trying to "refactor" the code to have it return earlier if not needed. I agree, the code is a mess (the PvP code is arguably the messiest I've worked with), so I highly recommend testing all the use cases here to make sure you aren't inadvertently breaking something else by moving the return statements around.

**Comment by RoboMWM** [comment]
And tbh, it'd be better to just fix/break down this handler instead of trying to achieve some microoptimizations (unless you do have profiles/samples that show significant improvements otherwise).

**Comment by FrankHeijden** [comment]
> Ok so I'm not sure if that's really what you're doing, as it looks more like you're just trying to "refactor" the code to have it return earlier if not needed.

This is exactly what I am trying to achieve here. When plugins cancel PreventPvPEvent, GP would not return there, but instead continue the rest of the methods code. This is something I want to prevent. I had cases where PreventPvPEvent was called twice, because GP doesn’t stop execution.

> And tbh, it'd be better to just fix/break down this handler instead of trying to achieve some microoptimizations

This is no feature or micro-optimization, it is a bugfix.

**Comment by RoboMWM** [comment]
Hmm ok. I see, pretty much all cases it's checking for it not being canceled, so the following would mean it is canceled.

I know why he used the negatory form for the event (easier to implement in existing code), but obviously not as easy to read, since multiple negatives have to be considered when trying to determine what the resulting behavior should be...

**Review by RoboMWM — APPROVED** [review]

**Pull request merged**